codice/ddf
# DDF-5215 Upgrade Solr to 7.7.2

Please review the Solr 7.x upgrade notes and verify I didn't miss anything: https://lucene.apache.org/solr/guide/7_7/solr-upgrade-notes.html#upgrading-to-7-x-releases

#### What does this PR do?
- Remove the sole usage of StandardFilterFactory as it is a no-op and was deprecated in Solr 7.5.
- Update docs to specify the correct Solr directories in which to place the jts-core and solr-xpath jars.
- Update the ddf-solr Docker image to copy the jts-core jar to the correct Solr directory.
- Update the solr-distro assembly file to copy the jts-core jar to the correct Solr directory.
- Delete solrconfig-inmemory.xml as it is unused.

#### Who is reviewing it? 
@leo-sakh 
@gjvera 

#### Select relevant component teams: 
@codice/solr 

#### Ask 2 committers to review/merge the PR and tag them here.
@troymohl 
@pklinef 
@rzwiefel 

#### How should this be tested?
1. Install DDF, ingest some records and test various textual, temporal and spatial queries.
2. Create and save several workspaces, search forms and result forms and verify they still behave as expected.
3. Repeat the above steps using a SolrCloud cluster (Solr 7.7.2 and ZooKeeper 3.4.14).

#### What are the relevant tickets?
Fixes: #5215 

#### Checklist:

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
